csharp/src/Microsoft.ML.OnnxRuntime/Training/CheckpointState.shared.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    ///  Holds the Checkpoint State as generated/consumed by on-device training APIs
+    /// </summary>
+    public class CheckpointState : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        /// <summary>
+        /// Creates CheckpointState by loading state from path.
+        /// <param name="checkpointPath"> absolute path to checkpoint file.</param>
+        /// </summary>
+        public CheckpointState(string checkpointPath)
```

Comment:
>CheckpointState [](http://example.com/codeflow?start=15&length=15)

Is this class being compiled in Inference only build? Can we make all of training code conditional if not already done?

Reply:
yes changing